### PR TITLE
Don't auto-join gated public channels

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: backend/canisters/community/impl/src/updates/c2c_join_channel.rs [hunk 1]
```diff
@@ -51,7 +51,16 @@ async fn c2c_join_channel(args: Args) -> Response {
     }
 }
 
-pub(crate) async fn join_channel_impl(channel_id: ChannelId, user_principal: Principal) -> Response {
+pub(crate) async fn join_channel_auto(channel_id: ChannelId, user_principal: Principal) {
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/updates/c2c_join_channel.rs:54]: This doesn't need to be async

**Review by hpeebles — DISMISSED** [review]

File: backend/canisters/community/impl/src/updates/c2c_join_channel.rs [hunk 2]
```diff
@@ -51,7 +51,16 @@ async fn c2c_join_channel(args: Args) -> Response {
...
+pub(crate) fn join_channel_auto(channel_id: ChannelId, user_principal: Principal) {
+    match read_state(|state| is_permitted_to_join(channel_id, user_principal, state)) {
+        Ok(None) => {}
+        _ => return,
+    };
+
+    mutate_state(|state| commit(channel_id, user_principal, state));
+}
+
+async fn join_channel_internal(channel_id: ChannelId, user_principal: Principal) -> Response {
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/updates/c2c_join_channel.rs:63]: Maybe this should be `check_gate_then_join_channel`?

Comment (hpeebles) [reply backend/canisters/community/impl/src/updates/c2c_join_channel.rs:63]: Its not very obvious what the difference is otherwise

Comment (megrogan) [reply backend/canisters/community/impl/src/updates/c2c_join_channel.rs:63]: ok

**Review by hpeebles — APPROVED** [review]

**Pull request merged**